restcomm/restcomm.interpreter/src/main/java/org/restcomm/connect/interpreter/VoiceInterpreter.java

```diff
@@ -2851,31 +2851,9 @@ public void execute(final Object message) throws Exception {
                         removeDialBranch(message, sender);
                         return;
                     } else {
-                        // At this point !sender.equal(call)
-                        // Ask the parser for the next action to take.
-                        Attribute attribute = null;
-                        if (verb != null) {
-                            attribute = verb.attribute("action");
+                        if (callState == CallStateChanged.State.IN_PROGRESS) {
+                            call.tell(new Hangup(), self());
                         }
-                        if (attribute == null) {
```

Comment:
it seems we removed quite a few lines here, where is all this stuff happening now?

I guess this is related to previous changes on the new COMPLETED condition to execute verbs.

Still is not clear to me the relationship? Could you elaborate the high level reasoning of the PR interms of the VI FSM? what the key changes in the states/transitions? what is the purpose of this change. All this explanation is just to faciliate myself understanding of the change(hopefully will help others as well)

Reply:
ive seen this note that partly answer my question:

> 

Note: RCML after Dial verb should still be executed (given there is no Dial Action attribute) but not for Dial, Play, Say, Gather, Record since the call is competed. SMS, Fax, Email should be executed ok.

Reply:
@jaimecasero The FSM will reach FinishDialing line 2854 https://github.com/RestComm/Restcomm-Connect/blob/21ff4940369a337f051b1ae2552a228e50b70a4b/restcomm/restcomm.interpreter/src/main/java/org/restcomm/connect/interpreter/VoiceInterpreter.java#L2854 when called client sends BYE. Before we were either executing Dial Action to get new RCML or were asking parser for next RCML but we didn't disconnected caller.
That means that in a sequential RMCL scenario:
1. A was connected to B, 
2. B was sending BYE
3. A was not disconnected but instead VI was executing next RCML verb
4. Thus, next Dial verb was executed, C was ringing

Now the patch disconnects A. When A completes, will send CallStateChanged.Completed to VI which will cause FSM to move to FinishDialing again, but this time will cleanup A call actor and will check next RCML.

Since CallState is Completed, subsequent Dial verbs will not be executed because of the changes in the `onTag()` method.